### PR TITLE
quic: iteration 13

### PR DESCRIPTION
**Pull request by jasnell**
This iteration is really just a number of cleanups plus some additional error handling.

##### Checklist

- [ ] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

### CODE REVIEW

File: src/node_quic_session.cc [hunk 1]
```diff
@@ -243,39 +247,58 @@ void QuicSession::AssociateCID(ngtcp2_cid* cid) {
...
+
+  HandleScope scope(env()->isolate());
+  Context::Scope context_scope(env()->context());
+
   Local<Value> argv[] = {
     Number::New(env()->isolate(), static_cast<double>(last_error.code)),
     Integer::New(env()->isolate(), last_error.family)
   };
+
+  // Grab a shared pointer to this to prevent the QuicSession
```
Comment (trivikr) [inline src/node_quic_session.cc]: Looks like this call is missed on line 994 https://github.com/nodejs/quic/blob/63b1b2f81e92b91ebdb777baad77a48fb715e640/src/node_quic_session.cc#L989-L995

Comment (jasnell) [reply src/node_quic_session.cc]: @trivikr ... please take a look again. The line numbers don't appear to be matching up so I'm not sure what you're referring to here.

Comment (trivikr) [reply src/node_quic_session.cc]: I searched for calls of `MakeCallback(` in `node_quic_session.cc` and noticed that there are 15 of them https://github.com/nodejs/quic/blob/master/src/node_quic_session.cc

And noticed that following line is added only on 14 of them in this PR
`std::shared_ptr<QuicSession> ptr(this->shared_from_this())`

Line number:
* nodejs:master is 994
* jasnell:iteration-13 is 1064
https://github.com/jasnell/quic/blob/3b8a9efce8640ce49365d74503528ed0d9b2c5fb/src/node_quic_session.cc#L1059-L1065

Comment (jasnell) [reply src/node_quic_session.cc]: Just did a quick scan through and verify that all of the MakeCallback calls in node_quic_session.cc include shared_ptr

**Comment by jasnell** [comment]
This PR should be ready to go

**Comment by jasnell** [comment]
Landed!

**Pull request closed without merging**